src/boot/efi/vmm.c

```diff
@@ -319,6 +339,54 @@ const char* smbios_find_oem_string(const char *name) {
         return NULL;
 }
 
+static const char* smbios_get_string(const SmbiosHeader *header, uint16_t string, uint64_t left) {
+        const char *s = (const void *) header;
+
+        s += header->length;
+        left -= header->length;
+
+        uint16_t index = 1;
+        for (const char *p = s; index <= string && p < s + left; index++) {
+                if (index == string)
+                    return p;
```

Comment:
indentation is off.

and i would prefer if we'd make sure that this is properly NUL terminated *before* we return it